### PR TITLE
Single-sourcing the package version, user-agent and bug fixes

### PR DESCRIPTION
**Pull request by fatihkurtoglu**
Summary of changes in separate commits as follows:

- Fixed project methods (naming typo in `create_project` and missing payload in `update_project`)
- Removed `build` folder from the repo. It's only required while building locally, similar to `dist` folder
- Enabled `User-Agent` parameter in request headers with module version, Python version and OS details. Sample:
```
scaleapi/1.0.2 Python/3.6.12 OS/Darwin-19.6.0-x86_64-i386-64bit
```

- Moved package versioning to a separate file: `_version.py` according to common practices:
  - `_version.py` will have a line for version in `__version__` variable
  - Same variable will be available within module as `scaleapi.__version__` as in most packages
  - Same value will be used in `setup.py` file
  - This approach is built according to one of the recommended practices in [Python Packaging User Guide](https://packaging.python.org/guides/single-sourcing-package-version/) 
 
Version bumped to `1.0.2` in file but will be built and uploaded to pypi after merge.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -25,10 +26,23 @@
...
+    with open(os.path.join(here, rel_path), 'r') as fp:
+        return fp.read()
+
+def get_version(rel_path):
+    for line in read(rel_path).splitlines():
+        if line.startswith('__version__'):
+            delim = '"' if '"' in line else "'"
+            return line.split(delim)[1]
+    else:
+        raise RuntimeError("Unable to find a valid __version__ string in %s." % rel_path)
```
Comment (shaun-scale) [inline setup.py:40]: Is this a cool python trick to put else: after a for loop or was the tab indentation just off here? I suppose we could just have it be `raise` without the `else:` to the same effect?

**Review by shaun-scale — CHANGES_REQUESTED** [review]
Amazing upgrades here @fatihkurtoglu - I had already bumped to v1.0.2 yesterday (sorry I wasn't clear there) and then I had one comment about our versioning logic but all the important stuff is looking great.

File: scaleapi/_version.py [hunk 2]
```diff
@@ -0,0 +1 @@
+__version__ = "1.0.2"
```
Comment (shaun-scale) [inline scaleapi/_version.py:1]: Oh not sure my comment here got saved, need to bump to `1.0.3`

File: setup.py (hunk 1 above)
Comment (fatihkurtoglu) [reply setup.py:40]: Yes, `for/else` is a pythonic way. `else` statement is triggered when the loop ends until the end without any interruption (break/return). More like a `then` stage for the loop. 

Yes, it's the same as just keeping the `raise` there as you said, I'll update it.

File: scaleapi/_version.py (hunk 2 above)
Comment (fatihkurtoglu) [reply scaleapi/_version.py:1]:
```suggestion
__version__ = "1.0.3"
```

**Review by fatihkurtoglu — COMMENTED** [review]
Version upgraded.

**Comment by fatihkurtoglu** [comment]
@shaun-scale Made the changes regarding version bump and for/else statement.

**Review by shaun-scale — APPROVED** [review]
Looks great

**Pull request merged**